x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/rank/textsimilarity/TextSimilarityRerankingRankFeaturePhaseRankShardContext.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.rank.textsimilarity;
+
+import org.elasticsearch.common.document.DocumentField;
+import org.elasticsearch.common.logging.HeaderWarning;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.SearchHits;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightBuilder;
+import org.elasticsearch.search.fetch.subphase.highlight.HighlightField;
+import org.elasticsearch.search.fetch.subphase.highlight.SearchHighlightContext;
+import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.search.rank.RankShardResult;
+import org.elasticsearch.search.rank.feature.RankFeatureDoc;
+import org.elasticsearch.search.rank.feature.RankFeatureShardResult;
+import org.elasticsearch.search.rank.feature.SnippetRankInput;
+import org.elasticsearch.search.rank.rerank.RerankingRankFeaturePhaseRankShardContext;
+import org.elasticsearch.xcontent.Text;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.search.rank.feature.RerankSnippetConfig.DEFAULT_NUM_SNIPPETS;
+
+public class TextSimilarityRerankingRankFeaturePhaseRankShardContext extends RerankingRankFeaturePhaseRankShardContext {
```

Comment:
Do we really need the separation here between `TextSimilarityRerankingRankFeaturePhaseRankShardContext` and `RerankingRankFeaturePhaseRankShardContext`? I don't see `RerankingRankFeaturePhaseRankShardContext` being used elsewhere so maybe we can just have `TextSimilarityRerankingRankFeaturePhaseRankShardContext` and implement the full logic there?

Reply:
> Can we just rename RerankingRankFeaturePhaseRankShardContext into TextSimilarityRerankingRankFeaturePhaseRankShardContext? I don't see what RerankingRankFeaturePhaseRankShardContext provides as an abstract class. 

@jimczi this class is referenced elsewhere including the random reranker. I personally think it's cleaner to have the separation, but if you feel strongly about it I will put snippet configuration in the parent class and remove the abstraction. I am commenting one more time instead of changing it, because I actually prefer this separation to putting things specific to text similarity that are used by the parent reranker. However, I won't die on that hill if you absolutely hate it.